packages/cli/src/workflows/workflows.controller.ts

```diff
@@ -396,6 +396,44 @@ export class WorkflowsController {
 		return true;
 	}
 
+	@Post('/:workflowId/archive')
+	@ProjectScope('workflow:delete')
+	async archive(req: WorkflowRequest.Delete) {
+		const { workflowId } = req.params;
+
+		const workflow = await this.workflowService.archive(req.user, workflowId);
+		if (!workflow) {
+			this.logger.warn('User attempted to archive a workflow without permissions', {
```

Comment:
If the workflow doesn't exist (as opposed to not being found due to permissions filters) then do we also return a nullish? If so then this message doesn't apply? Same below.

Reply:
We indeed return the same error if the workflow just doesn't exist. The new error message hopefully better applies here. I'm keeping the warning as is, users shouldn't be able to trigger this via UI.